### PR TITLE
[WIP] Add assert with relation

### PR DESCRIPTION
**Pull request by Jim-Holmstroem**
```
from workflow.testing.asserts import assert_

assert_.eq(1, 3)
```

### CODE REVIEW

File: workflow/testing/asserts.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        op.lt: '<',
+        op.gt: '>',
+        op.le: '<=',
+        op.ge: '>=',
+        op.contains: '∋',
+        _not_contains: '∌',
+    }[r]
+
+
+class assert_:
```
Comment (Jim-Holmstroem) [inline workflow/testing/asserts.py:42]: I don't like this myself, will move this stuff out to the module root instead
```
from workflow.testing import asserts
asserts.equal(1, 3)
```
is better

Comment (samedii) [reply workflow/testing/asserts.py:42]: Why do you prefer `assert.equal(1, 3)` over `assert 1 == 3`?

Edit: I saw now that you want to add a bunch of functionality

Comment (samedii) [reply workflow/testing/asserts.py:42]: I don't love this syntax:
```assert expression, 'my error message'```

I prefer
```
if not expression:
    raise AssertionError('my error message')
```

**Comment by samedii** [comment]
Try to rebase to new master (with ignite 0.3.0) and see if the tests pass

File: workflow/testing/asserts.py (hunk 1 above)
Comment (samedii) [reply workflow/testing/asserts.py:42]: @Jim-Holmstroem  Sorry I hijacked your comment by mistake (first wrote on mobile). Why is it not in the root right now? Did you run into problems?

Comment (Jim-Holmstroem) [reply workflow/testing/asserts.py:42]: I hate operators as well
we could wrap that as well here :)
#anticorruptionlayer

**Review by samedii — COMMENTED** [review]
- I did not love this syntax `["", f": {message}"][message is not None]`
- I find the assert syntax slightly harder to read

What do you think about something like this?
```
def _assert_relation(a, b, message=None):
    assert r(a, b), f"{a} {_render(_inverse(r))} {b}" + ["", f": {message}"][message is not None]
```
->
```
def _assert_relation(a, b, message=None):
    if message is None:
        postfix = ''
    else:
        postfix = f' : {message}'

    if not r(a, b):
        raise AssertionError(f"{a} {_render(_inverse(r))} {b}" + postfix)
```

**Comment by Jim-Holmstroem** [comment]
agree, an explicit if statement is easier to read, however I generally miss expression languages
`postfix = f' : {message}' if message else ''` which is in my opinion much more readable still.
I'll make the updates when the other things with higher priority is done :)

**Pull request closed without merging**